src/compiler/tsbuildPublic.ts

```diff
@@ -1568,42 +1733,68 @@ namespace ts {
         return actual;
     }
 
-    function updateOutputTimestampsWorker(state: SolutionBuilderState, proj: ParsedCommandLine, priorNewestUpdateTime: Date, verboseMessage: DiagnosticMessage, skipOutputs?: ESMap<Path, string>) {
-        if (proj.options.noEmit) return priorNewestUpdateTime;
+    function updateOutputTimestampsWorker(
+        state: SolutionBuilderState,
+        proj: ParsedCommandLine,
+        projectPath: ResolvedConfigFilePath,
+        verboseMessage: DiagnosticMessage,
+        skipOutputs?: ESMap<Path, string>
+    ) {
+        if (proj.options.noEmit) return;
+        let now: Date | undefined;
+        const buildInfoPath = getTsBuildInfoEmitOutputFilePath(proj.options);
+        if (buildInfoPath) {

```

Comment:
For incremental projects, only timestamp we need to update is for buildInfo if it wasnt emitted.